api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -588,6 +588,9 @@ message Cluster {
       // requests as if all hosts are unhealthy. This can help avoid potentially overwhelming a
       // failing service.
       bool fail_traffic_on_panic = 3;
+
+      // If set to true, Envoy will force LocalityDirect routing if a local locality exists.
+      bool force_direct_routing = 4;
```

Comment:
The new API in the extensions/common should be used. The legacy LB API here should be frozen.